src/pymmcore_plus/mda/_engine.py

```diff
@@ -285,10 +285,14 @@ def exec_single_event(self, event: MDAEvent) -> Iterator[PImagePayload]:
             return ()
         if not event.keep_shutter_open:
             self._mmc.setShutterOpen(False)
-        yield ImagePayload(self._mmc.getImage(), event, self.get_frame_metadata())
+
+        for cam in range(self._mmc.getNumberOfCameraChannels()):
+            data, meta = self._mmc.getTaggedImage(cam)
```

Comment:
i'd really like to avoid using `getTaggedImage` internally if possible.  It was added to mimic the java stuff, but it's very slow and exposes a ton of keys that are not really well defined (so I don't want to encourage people to use them yet).

Can you try to look deeper into the code, perhaps in `popNextImageAndMD` or in `getTags` and see if you can add the bare minimum needed to get_frame_metadata here?

I suspect it's these lines:
https://github.com/pymmcore-plus/pymmcore-plus/blob/e77d8301485daf9650a51cc6333f3b069e96cdf6/src/pymmcore_plus/core/_mmcore_plus.py#L1576-L1580